src/main/kotlin/no/nav/dagpenger/joark/mottak/JournalPostArkivHttpClient.kt

```diff
@@ -0,0 +1,52 @@
+package no.nav.dagpenger.joark.mottak
+
+import com.github.kittinunf.fuel.core.ResponseDeserializable
+import com.github.kittinunf.fuel.core.response
+import com.github.kittinunf.fuel.httpGet
+import com.github.kittinunf.result.Result
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.io.Reader
+import java.util.UUID
+
+class JournalPostArkivHttpClient(val joarkBaseUrl: String) : JournalpostArkiv {
+
+    override fun hentInngåendeJournalpost(journalpostId: String): JournalPost? {
+        val token = UUID.randomUUID().toString() //todo...
+        val url = "$joarkBaseUrl/rest/journalfoerinngaaende/v1/journalposter/$journalpostId"
+
+        val (_, response, result) = url.httpGet()
+                .header("Authorization" to token.toBearerToken())
+                .response(journalPostDeserializer())
```

Comment:
If you add the [GSON module](https://github.com/kittinunf/Fuel#gson-deserialization), you get out of the box automatic JSON deserializion.

Then something like this should work:
```
val (_, response, result)  = url.httpGet()
  .header("Authorization" to token.toBearerToken())
  .responseObject<JournalPost>()
```

And then the custom deserializer is not necessary.

Reply:
Then do not need / have any use of klaxon - but that is maybe ok?

Reply:
As a side note, idiomatic Kotlin always tries to avoid chaining. I guess a more Kotlinsk way of writing this would be using `with`. Minor detail though.

```
val (_, response, result)  = with(url.httpGet()) {
  header("Authorization" to token.toBearerToken())
  responseObject<JournalPost>()
}
```

Reply:
https://medium.com/@elye.project/mastering-kotlin-standard-functions-run-with-let-also-and-apply-9cd334b0ef84 is nice. Especially the flowchart in the bottom :)

Reply:
Added gson module and used `with`  :)